bin/lib/notify.py

```diff
@@ -0,0 +1,134 @@
+import urllib.request
+import urllib.parse
+import json
+from typing import List
+
+OWNER_REPO = "compiler-explorer/compiler-explorer"
+USER_AGENT = "CE Live Now Notification Bot"
+
+
+def post(entity: str, token: str, query: dict = None, dry=False) -> dict:
+    if query is None:
+        query = {}
+    path = entity
+    querystring = json.dumps(query).encode()
+    print(f"Posting {path}")
+    req = urllib.request.Request(
+        f"https://api.github.com/{path}",
+        data=querystring,
+        headers={
+            "User-Agent": USER_AGENT,
+            "Authorization": f"token {token}",
+            "Accept": "application/vnd.github.v3+json",
+        },
+    )
+    if dry:
+        return {}
+    result = urllib.request.urlopen(req)
+    # It's ok not to check for error codes here. We'll throw either way
+    return json.loads(result.read())
+
+
+def get(entity: str, token: str, query: dict = None) -> dict:
+    if query is None:
+        query = {}
+    path = entity
+    if query:
+        querystring = urllib.parse.urlencode(query)
+        path += f"?{querystring}"
+    print(f"Getting {path}")
+    req = urllib.request.Request(
+        f"https://api.github.com/{path}",
+        None,
+        {
+            "User-Agent": USER_AGENT,
+            "Authorization": f"token {token}",
+            "Accept": "application/vnd.github.v3+json",
+        },
+    )
+    result = urllib.request.urlopen(req)
+    # It's ok not to check for error codes here. We'll throw either way
+    return json.loads(result.read())
+
+
+def paginated_get(entity: str, token: str, query: dict = None) -> List[dict]:
+    if query is None:
+        query = {}
+    result: List[dict] = []
+    results_per_page = 50
+    query["page"] = 1
+    query["per_page"] = results_per_page
+    while True:
+        current_page_results = get(entity, token, query)
+        result.extend(current_page_results)
+        if len(current_page_results) == results_per_page:
+            query["page"] += 1
+        else:
+            break
+    return result
+
+
+def list_inbetween_commits(end_commit: str, new_commit: str, token: str) -> List[dict]:
+    commits = get(f"repos/{OWNER_REPO}/compare/{end_commit}...{new_commit}", token=token)
+    return commits["commits"]
+
+
+def get_linked_pr(commit: str, token: str) -> dict:
+    pr = get(f"repos/{OWNER_REPO}/commits/{commit}/pulls", token=token)
```

Comment:
Technically this is should be a paginated get, but if we find a PR with 30 or more linked issues.. Wow

Reply:
Also, the GraphQL query only lists the first 10, so there won't be more than that in the response, which fit in a single page

Reply:
Can graphql query on label status? I think that finding results that haven't been tagged `live` might be safer and easier on humans to find if it goes wrong, instead of having to look inside all the comments inside each PR